packages/sdk/src/utils/EditorResponseData.ts

```diff
@@ -1,17 +1,24 @@
-import { EditorResponse } from '../types/CommonTypes';
+import type { EditorResponse } from '../types/CommonTypes';
 
 export function getEditorResponseData<T>(response: EditorResponse<unknown>, parse = true): EditorResponse<T> {
-    return {
-        ...response,
-        parsedData:
-            response.success && response.data
-                ? parse
-                    ? (JSON.parse(response.data as string) as T)
-                    : (response.data as unknown as T)
-                : null,
-    };
+    try {
+        if (!response.success) {
+            throw new Error(response.error ?? 'Yikes, something went wrong');
```

Comment:
Can this throw a `SDKError` that has both `error`, `status` fields from `EditorResponse`?

Reply:
I rather wouldn't extend the error object so I made a small modificaiton to the return, can you have a look if this suits your needs?